packages/renderer/src/App.spec.ts

```diff
@@ -0,0 +1,78 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { render } from '@testing-library/svelte';
+import { tick } from 'svelte';
+import { router } from 'tinro';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import App from './App.svelte';
+
+const mocks = vi.hoisted(() => ({
+  DashboardPage: vi.fn(),
+  RunImage: vi.fn(),
+  ImagesList: vi.fn(),
+}));
+
+vi.mock('./lib/dashboard/DashboardPage.svelte', () => ({
+  default: mocks.DashboardPage,
+}));
+vi.mock('./lib/image/RunImage.svelte', () => ({
+  default: mocks.RunImage,
+}));
+vi.mock('./lib/image/ImagesList.svelte', () => ({
+  default: mocks.ImagesList,
+}));
+
+vi.mock('./lib/ui/TitleBar.svelte', () => ({
+  default: vi.fn(),
+}));
+vi.mock('./lib/welcome/WelcomePage.svelte', () => ({
+  default: vi.fn(),
+}));
+
+vi.mock('./lib/context/ContextKey.svelte', () => ({
+  default: vi.fn(),
+}));
+
+vi.mock('./lib/appearance/Appearance.svelte', () => ({
+  default: vi.fn(),
+}));
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  router.goto('/');
+});
+
+test('test /images/run/* route', async () => {
+  render(App);
+  expect(mocks.RunImage).not.toHaveBeenCalled();
+  expect(mocks.DashboardPage).toHaveBeenCalled();
+  router.goto('/images/run/basic');
+  await tick();
+  expect(mocks.RunImage).toHaveBeenCalled();
+});
+
+test('test /images/:id/:engineId route', async () => {
+  render(App);
+  expect(mocks.ImagesList).not.toHaveBeenCalled();
+  expect(mocks.DashboardPage).toHaveBeenCalled();
+  router.goto('/images/an-image/an-engine');
+  await tick();
+  expect(mocks.ImagesList).toHaveBeenCalled();
```

Comment:
tick is what id used in other tests inside `waitRender`. If I understand correctly, it is the time for Svelte to render the component again. waitFor seems more for unpredictable duration (with some callback reading disk, ...)

Reply:
I agree with @feloy 
I would say tick is the more accurate as you expect the screen to be updated by svelte, while for waitFor you wait for a condition

while at the end it may be the same result

Reply:
Okey thanks noted!